nengo/simulator.py

```diff
@@ -285,10 +291,18 @@ def run(self, time_in_seconds, progress_bar=None):
             For more control over the progress bar, pass in a `.ProgressBar`
             or `.ProgressUpdater` instance.
         """
-        steps = int(np.round(float(time_in_seconds) / self.dt))
-        logger.info("Running %s for %f seconds, or %d steps",
-                    self.model.label, time_in_seconds, steps)
-        self.run_steps(steps, progress_bar=progress_bar)
+        try:
+            steps = int(np.round(float(time_in_seconds) / self.dt))
+        except ZeroDivisionError:
+            steps = 0
+
+        if steps <= 0:
+            warnings.warn("0 timesteps requested. Simulator still at time %g."
+                          % (self.time,))
```

Comment:
It actually might make sense to raise an exception for `steps < 0` because running backwards is clearly wrong. (But not for `steps = 0` which might be intentional as running a Simulator for 0 steps should be a no-op. Though it could indicate a problem which warrants the warning.)

It might make sense to include the requested `time_in_seconds` in the message? That might help to see why 0 timesteps where requested ... (e.g., did I pass an actual 0s or was the duration > 0s, but too short?)

Reply:
The warning type should be a `NengoWarning`, maybe even a specific subtype to make it easy to silence the warning in programs where it is expected.

Reply:
No one mentioned this at the dev meeting and, in general, people were in favor of not stopping things, bothering the user, or changing existing behavior... I would be OK adding a different warning for negative values, but I feel like I would want other people to weigh in before adding an exception, and this PRs already dragged on for some time...

We raise lots of warnings that aren't `NengoWarning`. Unlike with exceptions, it's already easy to filter out warnings by module (see [filterwarnings](https://docs.python.org/3.5/library/warnings.html#warnings.filterwarnings); it has a `module` argument)... I can see the advantage of doing it, but not so much if we're not doing it consistently right now.

Reply:
> No one mentioned this at the dev meeting and, in general, people were in favor of not stopping things, bothering the user, or changing existing behavior...

I think we didn't discuss `steps < 0` explicitely and were more focussed on the rounding issue? Definitely wouldn't mind other people's opinion on this. I think it can be quite important to get the details right of what is an actual error condition, just a warning, or silently passes. But I am also somewhat traumatized by Matlab where too many things just silently pass (and various other software where an error is raised when things should silently pass).

I'm fine with leaving the warning type as `UserWarning` for now. It seems more discussion of what the best practices in that regard is required, but that doesn't need to block this PR as it isn't consistent in the code base right now anyways.